apps/condo/domains/common/components/GraphQlSearchInput/GraphQlSearchInput.tsx

```diff
@@ -60,28 +64,51 @@ export const GraphQlSearchInput: React.FC<ISearchInputProps> = (props) => {
         onSelect,
         formatLabel,
         renderOptions,
-        autoClearSearchValue,
+        autoClearSearchValue = true,
         initialValue,
         getInitialValueQuery,
         infinityScroll,
         disabled,
+        placeholder: initialPlaceholder,
+        value,
         eventName: propEventName,
         eventProperties = {},
         ...restProps
     } = props
+
+    const intl = useIntl()
+    const LoadingMessage = intl.formatMessage({ id: 'LoadingInProgress' })
+    const NotFoundMessage = intl.formatMessage({ id: 'NotFound' })
+
     const client = useApolloClient()
-    const [selected, setSelected] = useState('')
-    const [isLoading, setLoading] = useState(false)
-    const [data, setData] = useState([])
+    const [isSearchLoading, setSearchLoading] = useState(false)
+    const [isLoadingMore, setLoadingMore] = useState(false)
+    const [isInitialLoading, setInitialLoading] = useState(false)
     const [isAllDataLoaded, setIsAllDataLoaded] = useState(false)
-    const [value, setValue] = useState('')
+    const [initialData, setInitialData] = useState([])
+    const [allData, setAllData] = useState([])
+    const [options, setOptions] = useState([])
+    const [searchData, setSearchData] = useState([])
+    const [searchValue, setSearchValue] = useState('')
 
     const { logEvent, getEventName } = useTracking()
 
     const eventName = propEventName ? propEventName : getEventName(TrackingEventType.Select)
     const componentProperties = { ...eventProperties }
 
-    const renderOption = (option, index?) => {
+    const needShowLoadingMessage = isInitialLoading || ((isSearchLoading || isLoadingMore) && isEmpty(allData))
+    const placeholder = useMemo(() => needShowLoadingMessage ? LoadingMessage : initialPlaceholder,
+        [LoadingMessage, initialPlaceholder, needShowLoadingMessage])
+    const selectedValue = useMemo(() => needShowLoadingMessage ? [] : value,
+        [value, needShowLoadingMessage])
+    const isDisabled = disabled || needShowLoadingMessage
+
+    const notFoundContent = useMemo(() => isInitialLoading || isSearchLoading || isLoadingMore
+        ? <Loader size='small' delay={0} fill />
+        : NotFoundMessage,
+    [NotFoundMessage, isInitialLoading, isLoadingMore, isSearchLoading])
+
+    const renderOption = useCallback((option, index?) => {
         let optionLabel = option.text
```

Comment:
```
const optionLabel = isFunction(formatLabel) ? formatLabel(option) : option.text
```

Reply:
this function is specified in the interface